src/main/java/com/yscope/logging/log4j1/ClpIrFileAppender.java

```diff
@@ -0,0 +1,431 @@
+  private int compressionLevel = 3;
+  private boolean closeFrameOnFlush = true;
+  private String filePath;
+  private boolean useFourByteEncoding = false;
+
+  /**
+   * Default constructor (necessary for Log4j to instantiate the appender using
+   * reflection).
+   */
+  public ClpIrFileAppender () {
+  }
+
+  /**
+   * Constructs a ClpIrFileAppender
+   * @param filePath Output file path
+   * @param layout Log4j layout for formatting log events. Only
+   * {@code org.apache.log4j.EnhancedPatternLayout},
+   * {@code org.apache.log4j.PatternLayout}, and
+   * {@code org.apache.log4j.SimpleLayout} are supported.
+   * @param useFourByteEncoding Whether to use the four-byte encoding instead
+   * of the default eight-byte encoding
+   * @param closeFrameOnFlush Whether to close the Zstandard frame on flush
+   * @param compressionLevel Compression level to use for Zstandard
+   * @throws IOException on I/O error
+   */
+  public ClpIrFileAppender (
+      String filePath,
+      Layout layout,
+      boolean useFourByteEncoding,
+      boolean closeFrameOnFlush,
+      int compressionLevel
+  ) throws IOException {
+    super();
+
+    setFile(filePath);
+    setLayout(layout);
+    setUseFourByteEncoding(useFourByteEncoding);
+    setCompressionLevel(compressionLevel);
+    setCloseFrameOnFlush(closeFrameOnFlush);
+
+    super.activateOptions();
+
+    try {
+      validateOptionsAndInit();
+    } catch (Exception ex) {
+      closed = true;
+      throw ex;
+    }
+
+    activated = true;
+  }
+
+  // Public methods
+  public void setCompressionLevel (int compressionLevel) {
+    this.compressionLevel = compressionLevel;
+  }
+
+  public void setCloseFrameOnFlush (boolean closeFrameOnFlush) {
+    this.closeFrameOnFlush = closeFrameOnFlush;
+  }
+
+  public void setUseFourByteEncoding (boolean useFourByteEncoding) {
+    this.useFourByteEncoding = useFourByteEncoding;
+  }
+
+  public void setFile (String file) {
+    filePath = file;
+  }
+
+  public synchronized String getFile () {
+    return filePath;
+  }
+
+  /**
+   * @return The amount of data written to this appender for the current output
+   * file, in bytes.
+   * <p></p>
+   * NOTE:
+   * <ul>
+   *   <li>This may be slightly inaccurate since we use an estimate of the
+   *   timestamp length for performance reasons.</li>
+   *   <li>This will be reset when a new output file is opened.</li>
+   * </ul>
+   */
+  public synchronized long getUncompressedSize () {
+    return uncompressedSizeInBytes;
+  }
+
+  /**
+   * @return The amount of data written by this appender to the output file, in
+   * bytes.
+   */
+  public synchronized long getCompressedSize () {
+    return countingOutputStream.getByteCount();
+  }
+
+  /**
+   * Closes the previous file and starts a new file with the given path
+   * @param path
+   * @throws IOException on I/O error
+   */
+  public synchronized void startNewFile (String path) throws IOException {
+    if (false == activated) {
```

Comment:
nit: if !activated

Reply:
This is probably controversial, but our internal policy is to avoid using `!` unless absolutely necessary. Reason being, it's easy to miss when reading the code, especially when a developer is inevitably coding while tired.

Reply:
ACK